wordpress-mobile/WordPress-iOS
# [Blog Details] Make Site Title tappable

Fixes #14361. This PR allows the site title to be changed by tapping on the site title in the blog details view.

![site-title-change](https://user-images.githubusercontent.com/4780/85908212-1c2c3900-b80c-11ea-8b2e-5e7b94fd6962.gif)

This also hooks up the Quick Start flow implemented in #14384 to use the new method of setting the title.

Note that the flow here differs from the one set out in the original issue. The flow described there turned out to be difficult to implement and was a confusing user experience, particularly on iPad. This new simpler flow is closer to the updated flow being [implemented on Android](https://github.com/wordpress-mobile/WordPress-Android/pull/12269).

* Tapping the site title shows a site settings modal which allows you to cancel or to change the title and save
* On cancel, nothing happens
* On save, the site title will be updated in the background. On successful save, a notice will be displayed and the title will be refreshed in the header view
* If the user isn't an admin, different prompt text will be displayed below the site title text field in the settings screen. If the user tries to save it, a notice will be displayed containing the same text.

**To test:**

* Build and run – please test on iPhone and iPad
* As an admin user, navigate to a site and tap the site's title on the blog details screen. Ensure you can cancel out of the modal without changing the title, and that you can change the title and save your changes.
* Ensure that the site header updates to reflect your new title
* Activate Quick Start for the site (you can do this via the Me menu > App Settings > Debug (at the bottom) > Enable quick start for site
* Walk through the "Set a site title" Quick Start tour and ensure that a spotlight indicator appears on the site title
* Tap the site title to change it and ensure the indicator disappears
* In the Quick Start screen, check that the tour was ticked off on the list

PR submission checklist:
